### PR TITLE
Update templates to address defeciency and suggestion from 2021 audit

### LINKED ISSUE #18816: D-01/S-02: Include all sections in app/modules SQA templates
## Reason

In an NQA-1 assessment a deficiency (D-01) was noted that our SQA templates did not include all the necessary sections as defined by INL.

Suggestion S-02 adds that within the RTM, if no requirements exist that there should be some text explaining that rather than an empty section.

## Design

The app/modules SQA templates should include all the necessary sections a defined by INL. The default should link to the framework level document, with the option to override it.

If no requirements exist with in a section add a statement automatically that for this application there is no reqs. of this type.

## Impact

Applications will need to ensure that the default template documentation is accurate for their application.

### PR DESCRIPTION
**Pull request by aeslaughter**
Closes D-01, D-05, S-02, S-05 from 2021 audit.

(closes #18816)

### CODE REVIEW

**Review by cticenhour — CHANGES_REQUESTED** [review]
Just a couple changes and comments.

File: modules/stochastic_tools/doc/content/modules/stochastic_tools/sqa/stochastic_tools_srs.md [hunk 1]
```diff
@@ -1 +1,6 @@
 !template load file=sqa/app_srs.md.template category=stochastic_tools app=Stochastic Tools
+
+
+!template! item key=system-purpose
+STM does stuff with stuff.
```
Comment (aeslaughter) [inline modules/stochastic_tools/doc/content/modules/stochastic_tools/sqa/stochastic_tools_srs.md:5]:
```suggestion
!template load file=sqa/app_srs.md.template category=stochastic_tools app=Stochastic Tools
```

[the remark this answers is not in the document]
Comment (aeslaughter) [reply modules/stochastic_tools/doc/content/modules/stochastic_tools/sqa/stochastic_tools_srs.md:5]: I will double check this behavior and fix my little test there.

[the remark this answers is not in the document]
Comment (aeslaughter) [reply modules/stochastic_tools/doc/content/modules/stochastic_tools/sqa/stochastic_tools_srs.md:5]: You are correct about the behavior. That page exists for the build of the stochastic_tools site by itself, in that case those links point to the stochastic_tool versions. 

You can get to the module RTM, SRS, etc. via the list in the dependencies section of the MOOSE level documents. But, that isn't obvious so I am going to rework the template a bit to put those links up at the top.

[the remark this answers is not in the document]
Comment (cticenhour) [reply modules/stochastic_tools/doc/content/modules/stochastic_tools/sqa/stochastic_tools_srs.md:5]: Sounds good - are you going to include those changes in this PR, or a follow-on?

**Review by cticenhour — APPROVED** [review]

**Pull request merged**